app/src/main/java/com/android/solvit/provider/ui/request/ListRequestsFeed.kt

```diff
@@ -0,0 +1,434 @@
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.ColorFilter
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.lifecycle.viewmodel.compose.viewModel
+import coil.compose.AsyncImage
+import com.android.solvit.R
+import com.android.solvit.seeker.ui.navigation.SeekerBottomNavigationMenu
+import com.android.solvit.shared.model.map.Location
+import com.android.solvit.shared.model.request.ServiceRequest
+import com.android.solvit.shared.model.request.ServiceRequestViewModel
+import com.android.solvit.shared.model.service.Services
+import com.android.solvit.shared.ui.map.GetDirectionsBubble
+import com.android.solvit.shared.ui.navigation.LIST_TOP_LEVEL_DESTINATION_PROVIDER
+import com.android.solvit.shared.ui.navigation.NavigationActions
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+@Preview
+@Composable
+fun RequestsTopBar() {
+  val context = LocalContext.current
+  Row(
+      modifier = Modifier.fillMaxWidth().background(Color.White).testTag("RequestsTopBar"),
+      horizontalArrangement = Arrangement.SpaceBetween,
+      verticalAlignment = Alignment.CenterVertically) {
+        IconButton(
+            modifier = Modifier.testTag("MenuOption"),
+            onClick = {
+              Toast.makeText(context, "Not Yet Implemented", Toast.LENGTH_LONG).show()
+            }) {
+              Icon(imageVector = Icons.Default.Menu, contentDescription = "Menu Option")
+            }
+
+        Row(
+            modifier = Modifier.testTag("SloganIcon"),
+            verticalAlignment = Alignment.CenterVertically,
+        ) {
+          Text(
+              text = "Solv",
+              style =
+                  TextStyle(
+                      fontSize = 20.sp, fontWeight = FontWeight.Bold, color = Color(0xFF333333)))
+          Text(
+              text = "It",
+              style =
+                  TextStyle(
+                      fontSize = 20.sp, fontWeight = FontWeight.Bold, color = Color(0xFF3D823B)))
+        }
+
+        IconButton(
+            onClick = {
+              Toast.makeText(context, "Not Yet Implemented", Toast.LENGTH_LONG).show()
+            }) {
+              Icon(imageVector = Icons.Default.Notifications, contentDescription = "Notifications")
+            }
+      }
+}
+
+@Composable
+fun SearchBar(searchQuery: MutableState<String>) {
```

Comment:
Nit: Same comment as above. We use a search bar multiple times throughout the project, it could be refactored.

Reply:
You are right!